tensorflow/python/ops/math_grad.py

```diff
@@ -411,6 +417,16 @@ def _SigmoidGrad(op, grad):
     return gen_math_ops._sigmoid_grad(y, grad)
 
 
+@ops.RegisterGradient("SigmoidGrad")
+def _SigmoidGradGrad(op, grad):
+  y = op.outputs[0]
+  with ops.control_dependencies([grad.op]):
+    if y.dtype.is_complex:
+      y = math_ops.conj(y)
+    return (- math.ops.exp(x) * (math.ops.exp(x) - 1) *
+            math_ops.pow(math.ops.exp(x) + 1, -3))
```

Comment:
@girving Yes, I missed the grad and the math.ops instead of math_ops.  Is
there a reason you associated the grad with s? or would you like grad * s *
(1 - s) * (1 - 2 * x)?

I will add the tests into math_grad_test.py

On Sun, Aug 14, 2016 at 4:58 PM, Geoffrey Irving <notifications@github.com>
wrote:

> In tensorflow/python/ops/math_grad.py
> <https://github.com/tensorflow/tensorflow/pull/3807#discussion_r74711318>:
>
> > @@ -411,6 +417,16 @@ def _SigmoidGrad(op, grad):
> >      return gen_math_ops._sigmoid_grad(y, grad)
> >
> >
> > +@ops.RegisterGradient("SigmoidGrad")
> > +def _SigmoidGradGrad(op, grad):
> > +  y = op.outputs[0]
> > +  with ops.control_dependencies([grad.op]):
> > +    if y.dtype.is_complex:
> > +      y = math_ops.conj(y)
> > +    return (- math.ops.exp(x) * (math.ops.exp(x) - 1) *
> > +            math_ops.pow(math.ops.exp(x) + 1, -3))
>
> This explodes for large x and has the downside of being a constant
> function of grad (unusual for a gradient, so maybe a test is in order).
> Instead, do
>
> s = math_ops.sigmoid(x)
> return (grad * s) * (1 - s) * (1 - 2 * x)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/3807/files/253c86d7783dca1c844ff77afad5342c40691e66#r74711318>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADzDDDQKqJ_DtWtjp5PJSc98VuY_JlfMks5qf6uugaJpZM4Jj8mC>
> .
>